### PR TITLE
feat: usage analytics report to azure application insights

### PR DESCRIPTION
**Pull request by veredcon**

### CODE REVIEW

File: packages/backend/src/usage-report/usage-analytics-wrapper.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  public static createTracker(logger?: IChildLogger) {
+    try {
+      const packageJson = JSON.parse(fs.readFileSync(path.join(__dirname, "..", "package.json"), "utf8"));
+      const vscodeExtentionFullName = `${packageJson.publisher}.${packageJson.name}`;
+      initTelemetrySettings(vscodeExtentionFullName, packageJson.version);
+      if (logger) {
+        logger.info(`SAP Web Analytics tracker was created for ${vscodeExtentionFullName}`);
+      }
+    } catch (error) {
+      logger.error(error);
```
Comment (alex-gilin) [inline packages/backend/src/usage-report/usage-analytics-wrapper.ts:38]:
```suggestion
      logger?.error(error);
```

File: packages/backend/src/usage-report/usage-analytics-wrapper.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      const endTime = Date.now();
+      const generationTimeMilliSec = endTime - AnalyticsWrapper.startTime;
+      const generationTimeSec = Math.round(generationTimeMilliSec / 1000);
+      const properties: any = {
+        generatorName,
+        generationTime: generationTimeSec.toString(),
+      };
+      AnalyticsWrapper.report(eventName, properties, logger);
+    } catch (error) {
+      logger.error(error);
```
Comment (alex-gilin) [inline packages/backend/src/usage-report/usage-analytics-wrapper.ts:89]:
```suggestion
      logger?.error(error);
```

File: packages/backend/src/usage-report/usage-analytics-wrapper.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    }
+  }
+
+  private static report(eventName: string, properties?: any, logger?: IChildLogger) {
+    // We want to report only if we are not in Local VSCode environment
+    if (process.env.LANDSCAPE_ENVIRONMENT) {
+      void AnalyticsWrapper.getTracker().report(eventName, properties);
+      logger?.trace("SAP Web Analytics tracker was called and start time was initialized", {
```
Comment (alex-gilin) [inline packages/backend/src/usage-report/usage-analytics-wrapper.ts:46]: is the log text as intended - is there a case where the start time is NOT initialized?

File: packages/backend/.nycrc.json [hunk 4]
```diff
@@ -24,5 +24,5 @@
   "branches": 91,
   "lines": 93,
   "functions": 88,
-  "statements": 93
+  "statements": 92
```
Comment (alex-gilin) [inline packages/backend/.nycrc.json:27]: 👎

**Review by alex-gilin — CHANGES_REQUESTED** [review]

File: packages/backend/src/usage-report/usage-analytics-wrapper.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
+import { initTelemetrySettings, BASClientFactory, BASTelemetryClient } from "@sap/swa-for-sapbas-vsx";
+import { IChildLogger } from "@vscode-logging/logger";
+import * as fs from "fs";
+import * as path from "path";
```
[the remark this answers is not in the document]
Comment (veredcon) [reply packages/backend/src/usage-report/usage-analytics-wrapper.ts:4]: solved together with require

File: packages/backend/webpack.config.js [hunk 6]
```diff
@@ -37,6 +37,33 @@ const config = {
...
+      },
+      {
+        test: /yeoman-environment[/|\\]lib[/|\\]environment.js/,
+        loader: "string-replace-loader",
+        options: {
+          search: "require[.]resolve[(]",
+          replace: "__non_webpack_require__.resolve(",
+          flags: "g",
+        },
+      },
```
Comment (alex-gilin) [inline packages/backend/webpack.config.js:66]: looks like added by mistake. please verify

**Review by alex-gilin — CHANGES_REQUESTED** [review]

**Review by alex-gilin — APPROVED** [review]

**Pull request merged**